packages/opentelemetry-metrics/src/MeterProvider.ts

```diff
@@ -14,27 +14,43 @@
  * limitations under the License.
  */
 
-import { ConsoleLogger } from '@opentelemetry/core';
 import * as api from '@opentelemetry/api';
+import { ConsoleLogger } from '@opentelemetry/core';
+import { Meter } from './Meter';
+import { MetricRecord } from './export/types';
+import { Batcher, UngroupedBatcher } from './export/Batcher';
 import { Resource } from '@opentelemetry/resources';
-import { Meter } from '.';
-import { DEFAULT_CONFIG, MeterConfig } from './types';
+import { MeterConfig, DEFAULT_CONFIG } from './types';
+import { Controller } from './export/controllers';
 
 /**
  * This class represents a meter provider which platform libraries can extend
  */
 export class MeterProvider implements api.MeterProvider {
   private readonly _config: MeterConfig;
   private readonly _meters: Map<string, Meter> = new Map();
+  private readonly _batcher: Batcher;
   readonly resource: Resource;
   readonly logger: api.Logger;
 
   constructor(config: MeterConfig = DEFAULT_CONFIG) {
     this.logger = config.logger ?? new ConsoleLogger(config.logLevel);
     this.resource = config.resource ?? Resource.createTelemetrySDKResource();
+    this._batcher = config.batcher ?? new UngroupedBatcher();
     this._config = Object.assign({}, config, {
       logger: this.logger,
       resource: this.resource,
+      batcher: this._batcher,
+    });
+  }
+
+  /**
+   * Adds a new {@link Controller} to this meter provider.
+   * @param controller the new Controller to be added.
+   */
+  addController(controller: Controller) {
+    controller.registerMetricsCollector({
+      collect: () => this._collect(),
```

Comment:
You can avoid the extra function call and associated overhead if you make the `collect` method public and do this.

```typescript
controller.registerMetricsCollector(this);
```

Reply:
Or you could just inline the collect function implementation since it is only called in one place.

Reply:
Fixed.